autoload/ale/completion.vim

```diff
@@ -998,6 +1001,10 @@ function! ale#completion#Queue() abort
 
     call ale#completion#StopTimer()
 
+    if !empty(complete_info(['mode']))
```

Comment:
This line does not work at all.  That complete_info method does not exists. See CI errors:

```
Starting Vader: C:\testplugin\test\completion\test_completion_events.vader
    ( 1/29) [EXECUTE] ale#completion#GetCompletions should be called when the cursor position stays the same
    ( 1/29) [EXECUTE] (X) Vim(if):E117: Unknown function: complete_info
      > function CheckCompletionCalled[12]..ale#completion#Queue, line 21
    ( 2/29) [EXECUTE] ale#completion#GetCompletions should not be called if the global setting is disabled
    ( 3/29) [EXECUTE] ale#completion#GetCompletions should not be called if the buffer setting is disabled
    ( 4/29) [  GIVEN]
    ( 4/29) [EXECUTE] ale#completion#GetCompletions should not be called when the cursor position changes
    ( 4/29) [EXECUTE] (X) Vim(if):E117: Unknown function: complete_info
      > function ale#completion#Queue, line 21
    ( 5/29) [  GIVEN]
    ( 5/29) [EXECUTE] ale#completion#GetCompletions should not be called if you switch to normal mode
    ( 5/29) [EXECUTE] (X) Vim(if):E117: Unknown function: complete_info
      > function ale#completion#Queue, line 21
    ( 6/29) [  GIVEN]
    ( 6/29) [EXECUTE] Completion should not be done shortly after the CompleteDone function
    ( 6/29) [EXECUTE] (X) Vim(if):E117: Unknown function: complete_info
      > function CheckCompletionCalled[12]..ale#completion#Queue, line 21
    ( 7/29) [  GIVEN]
```

Reply:
@hsanson I ...am at a loss here? 

:help complete_info on VIM 8.2.4793 (this is whatever debian packages) comes up with the proper builtin function documentation. Only disclaimer, I have no idea when this function was introduced.

I will have to look into and understand the testing process for ALE, but in the mean time, what versions of VIM do these tests run against?

Reply:
This function does not exists in neovim 0.2 and vim 8.0 that are still supported by ALE.

We would need to figure out a way to make this work on all supported versions. Like creating a function that checks for the existence of that method before using it:

```
function ale#completion#Info(what)
    if exists('*complete_info')
        return complete_info(a:what)
    else
       // Some code that does same as complete_info 
       // in older version of vim and neovim
    endif
endfunction
```

Reply:
That's unfortunate. I'll have a look into vim 8 and see if there's an alternative way to achieve the same thing, but it's not urgent for me. I can just keep rebasing my fork. Thanks for the pointers!

Reply:
Spun up a couple of containers and looked into vim 8.0 and 8.1. Surprisingly to me, ```complete_info()``` and CTRL^X CTRL^Z (to exit autocompletion) are only available to vim 8.2 (did not check neovim at all).

An alternative would be to check ```mode()``` and if it is 'i' then issue CTRL^E to stop completion. I'll be looking into that a bit later.

In general, what's the stance of this project when it comes to using features  introduced in later versions? Is this if-else pattern common? Or for example completely dropping the else part when the functionality cannot be reproduced in earlier versions?

Reply:
If omiting the else does not break anything then I would say is ok to omit. In worst case something will work better on newer versions of vim/neovim and nothing happens on older versions.

Reply:
So, I did implement the conditional logic, looking specifically to see if ```complete_info()``` is available, and if it's not, it'll be a no-op. 

As far as I can tell, we'll get the desired functionality in vim8.2 and no changes in older versions.

As a side-note, I did try using ```mode()``` and CTRL^E for older versions of vim, and it *almost* worked: autocompletion behaved nicely in vim8.0 *but* it had side effects when I was editing other stuff (text appeared when editing comments, etc).